# 0. import 라이브러리 


In [1]:
import numpy as np
import pandas as pd

# 1. 데이터 준비

## 1.1 데이터 불러오기

In [2]:
# 데이터 불러오기
recruitment_data = pd.read_csv("./data/raw_data/recruitment.csv")
company_data = pd.read_csv("./data/raw_data/company.csv")
resume_data = pd.read_csv("./data/raw_data/resume.csv")
resume_certificate_data = pd.read_csv("./data/raw_data/resume_certificate.csv")
resume_education_data = pd.read_csv("./data/raw_data/resume_education.csv")
resume_language_data = pd.read_csv("./data/raw_data/resume_language.csv")

apply_train_data = pd.read_csv("./data/raw_data/apply_train.csv")
sample_submission_data = pd.read_csv("./data/raw_data/sample_submission.csv")

## 1.2 데이터 확인 및 정제

### 1.2.1 recruitment (6,695)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|recruitment_seq|공고번호|PK|
|1|address_seq1|근무지주소코드1||
|2|address_seq2|근무지주소코드2||
|3|address_seq3|근무지주소코드3||
|4|career_end|요구 최대경력||
|5|career_start|요구 최소경력||
|6|check_box_keyword|모집직무코드|; 으로 구분|
|7|education|요구학위||
|8|major_task|주업무코드||
|9|qualifications|자격요건난이도||
|10|text_keyword|모집직무키워드|; 으로 구분|

</details>

In [3]:
# 데이터 확인
display(recruitment_data.tail(3))

# 결측치 확인
print(recruitment_data.isna().sum())

recruitment_seq  address_seq1  address_seq2  address_seq3  career_end  \
6692          R03252           3.0           NaN           NaN           0   
6693          R05130           3.0           NaN           NaN           0   
6694          R04779           3.0           NaN           NaN           0   

      career_start    check_box_keyword  education  major_task  \
6692             0                 2109          3           2   
6693             0  2201;2204;2205;2707          2           2   
6694             0  2101;2108;2201;2707          3           8   

      qualifications text_keyword  
6692               1          NaN  
6693               2          NaN  
6694               1          NaN

recruitment_seq         0
address_seq1            1
address_seq2         6595
address_seq3         6686
career_end              0
career_start            0
check_box_keyword       0
education               0
major_task              0
qualifications          0
text_keyword         5988
dtype: int64


In [4]:
# address_seq 컬럼 중 NaN를 0.0로 변경
recruitment_data.address_seq1 = recruitment_data.address_seq1.replace(np.NAN, 0.0)
recruitment_data.address_seq2 = recruitment_data.address_seq2.replace(np.NAN, 0.0)
recruitment_data.address_seq3 = recruitment_data.address_seq3.replace(np.NAN, 0.0)

# text_keyword 컬럼 중 NaN를 'NaN'로 변경
recruitment_data.text_keyword = recruitment_data.text_keyword.replace(np.NAN, 'NaN')

# 결측치 재확인
print(recruitment_data.isna().sum())

recruitment_seq      0
address_seq1         0
address_seq2         0
address_seq3         0
career_end           0
career_start         0
check_box_keyword    0
education            0
major_task           0
qualifications       0
text_keyword         0
dtype: int64


### 1.2.2 company (2,377)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|recruitment_seq|공고번호|FK|
|1|company_type_seq|회사유형코드||
|2|supply_kind|주업종코드||
|3|employee|종업원수||
<details>

In [5]:
# 데이터 확인
display(company_data.tail(3))

# 결측치 확인
print(company_data.isna().sum())

recruitment_seq  company_type_seq  supply_kind  employee
2374          R04028                 4          402       525
2375          R06508                 2          402        70
2376          R00604                 5          402       300

recruitment_seq     0
company_type_seq    0
supply_kind         0
employee            0
dtype: int64


### 1.2.3 resume (8,482)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|PK|
|1|reg_data|이력서등록일||
|2|updated_data|이력서수정일||
|3|degree|최종학력||
|4|graduate_data|졸업년도||
|5|hope_salary|희망연봉||
|6|last_salary|최종연봉||
|7|text_keyword|직무키워드|; 으로 구분|
|8|job_code_seq1|희망직무1||
|9|job_code_seq2|희망직무2||
|10|job_code_seq3|희망직무3||
|11|career_month|경력월||
|12|career_job_code|경력직무|; 으로 구분|
</details>

In [6]:
# 데이터 확인
display(resume_data.tail(3))

# 결측치 확인
print(resume_data.isna().sum())

resume_seq    reg_date updated_date  degree  graduate_date  hope_salary  \
8479     U01468  2019-02-26   2020-06-23       4           2008          0.0   
8480     U05315  2020-03-27   2020-12-21       4           2016          0.0   
8481     U01277  2020-09-22   2020-09-22       4           2009       2100.0   

      last_salary          text_keyword job_code_seq1 job_code_seq2  \
8479          0.0          상품기획MD;우븐디자인           디자인           NaN   
8480          0.0  여성복디자인;우븐디자인;다이마루디자인   재료·화학·섬유·의복           NaN   
8481       1000.0            상품기획;머천다이저   재료·화학·섬유·의복           NaN   

     job_code_seq3  career_month career_job_code  
8479           NaN             0     섬유;봉제;가방;의류  
8480           NaN            18      디자인;CAD;설계  
8481           NaN             3     섬유;봉제;가방;의류

resume_seq            0
reg_date              0
updated_date          0
degree                0
graduate_date         0
hope_salary           0
last_salary           0
text_keyword        195
job_code_seq1         0
job_code_seq2      7982
job_code_seq3      8295
career_month          0
career_job_code     758
dtype: int64


In [7]:
# text_keyword 컬럼 중 NaN를 'NaN'로 변경
resume_data.text_keyword = resume_data.text_keyword.replace(np.NAN, 'NaN')

# job_code_seq 컬럼 중 NaN를 'NaN'로 변경
resume_data.job_code_seq2 = resume_data.job_code_seq2.replace(np.NAN, 'NaN')
resume_data.job_code_seq3 = resume_data.job_code_seq3.replace(np.NAN, 'NaN')

# career_job_code 컬럼 중 NaN를 'NaN'로 변경
resume_data.career_job_code = resume_data.career_job_code.replace(np.NAN, 'NaN')


# 결측치 재확인
print(recruitment_data.isna().sum())

recruitment_seq      0
address_seq1         0
address_seq2         0
address_seq3         0
career_end           0
career_start         0
check_box_keyword    0
education            0
major_task           0
qualifications       0
text_keyword         0
dtype: int64


### 1.2.4 resume_certificate (12,975)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|FK|
|1|certificate_contents|자격증명||
</details>

In [8]:
# 데이터 확인
display(resume_certificate_data.tail(3))

# 결측치 확인
print(resume_certificate_data.isna().sum())

resume_seq certificate_contents
12972     U03248            수치제어밀링기능사
12973     U08157                 주산2급
12974     U02889             유통관리사 2급

resume_seq              0
certificate_contents    4
dtype: int64


In [9]:
# certificate_contents 컬럼 중 NaN를 'NaN'로 변경
resume_certificate_data.certificate_contents = resume_certificate_data.certificate_contents.replace(np.NAN, 'NaN')


# 결측치 재확인
print(resume_certificate_data.isna().sum())

resume_seq              0
certificate_contents    0
dtype: int64


### 1.2.5 resume_education (8,482)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|FK|
|1|hischool_type_seq|고등학교유형코드||
|2|hischool_special_type|특수고등학교분류||
|3|hischool_nation|고등학교공립/사립구분||
|4|hischool_gender|고등학교남녀/공학구분||
|5|hischool_location_seq|고등학교지역코드||
|6|univ_type_seq1|최종대학(원)유형코드1||
|7|univ_type_seq2|최종대학(원)유형코드2||
|8|univ_tranfer|최종대학(원)편입여부|편입: 1|
|9|univ_location|최종대학(원)지역코드|
|10|univ_major|최종대학(원)전공||
|11|univ_sub_major|최종대학(원)부전공||
|12|univ_major_type|최종대학(원)주전공계열코드||
|13|univ_score|최종대학(원)평균평점||
</details>

In [10]:
# 데이터 확인
display(resume_education_data.tail(3))

# 결측치 확인
print(resume_education_data.isna().sum())

resume_seq  hischool_type_seq hischool_special_type hischool_nation  \
8479     U04109                 21                   일반고              공립   
8480     U01863                 21                   일반고              사립   
8481     U05884                  0                    기타              기타   

     hischool_gender  hischool_location_seq  univ_type_seq1  univ_type_seq2  \
8479            남여공학                      5               0               0   
8480            여자학교                     15               3               5   
8481              기타                      0               5               5   

      univ_transfer  univ_location  univ_major univ_sub_major  \
8479              0              0         NaN            NaN   
8480              0             15  의류학과생활과학계열            NaN   
8481              0             11         NaN            NaN   

      univ_major_type  univ_score  
8479                9        80.0  
8480                9        80.0  
8481                9        90.0

resume_seq                  0
hischool_type_seq           0
hischool_special_type       0
hischool_nation             0
hischool_gender             0
hischool_location_seq       0
univ_type_seq1              0
univ_type_seq2              0
univ_transfer               0
univ_location               0
univ_major               6687
univ_sub_major           8066
univ_major_type             0
univ_score                  0
dtype: int64


In [11]:
# certificate_contents 컬럼 중 NaN를 'NaN'로 변경
resume_education_data.univ_major  = resume_education_data.univ_major .replace(np.NAN, 'NaN')

# certificate_contents 컬럼 중 NaN를 'NaN'로 변경
resume_education_data.univ_sub_major  = resume_education_data.univ_sub_major.replace(np.NAN, 'NaN')


# 결측치 재확인
print(resume_education_data.isna().sum())

resume_seq               0
hischool_type_seq        0
hischool_special_type    0
hischool_nation          0
hischool_gender          0
hischool_location_seq    0
univ_type_seq1           0
univ_type_seq2           0
univ_transfer            0
univ_location            0
univ_major               0
univ_sub_major           0
univ_major_type          0
univ_score               0
dtype: int64


### 1.2.6 resume_language (869)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|FK|
|1|language|외국어코드||
|2|exam_name|공인시험코드||
|3|score|점수||
</details>

In [12]:
# 데이터 확인
display(resume_language_data.tail(3))

# 결측치 확인
print(resume_language_data.isna().sum())

resume_seq  language  exam_name   score
866     U05922         2          2  500.00
867     U03827         2          4  883.84
868     U04680         2          4  666.67

resume_seq    0
language      0
exam_name     0
score         0
dtype: int64


### 1.2.7 apply_train (57,946)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|FK|
|<span style="color:yellow">1|<span style="color:yellow">recruitment_seq|<span style="color:yellow">공고번호|<span style="color:yellow">TARGET(FK)</span>|
</deatils>

In [13]:
# 데이터 확인
display(apply_train_data.tail(3))

# 결측치 확인
print(apply_train_data.isna().sum())

resume_seq recruitment_seq
57943     U08238          R01342
57944     U01296          R06363
57945     U05748          R03090

resume_seq         0
recruitment_seq    0
dtype: int64


### 1.2.8 sample_submission (42,410)

> 사용자별로 5개씩 공고ID추천리스트 생성

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|FK|
|<span style="color:yellow">1|<span style="color:yellow">recruitment_seq|<span style="color:yellow">공고번호|<span style="color:yellow">TARGET(FK)</span>|
</details>

In [14]:
# 데이터 확인
display(sample_submission_data.tail(3))

# 결측치 확인
print(sample_submission_data.isna().sum())

resume_seq recruitment_seq
42407     U08482          R01931
42408     U08482          R01607
42409     U08482          R06059

resume_seq         0
recruitment_seq    0
dtype: int64


## 1.3 데이터 전처리

### 1.3.1 채용 공고 관련 데이터 중 사용할 데이터 추출

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|example|note|
|--|--|--|--|--|
|0|<span style="color:orange">recruitment_seq|<span style="color:orange">공고번호|R02444|PK|
|1|<span style="color:orange">address_seq1|<span style="color:orange">근무지주소코드1|3.0||
|2|<span style="color:orange">address_seq2|<span style="color:orange">근무지주소코드2|20.0||
|3|address_seq3|근무지주소코드3|20.0|기입된 데이터는 9개 뿐으로 사용 X|
|4|career_end|요구 최대경력|0|모든 값이 0|
|5|career_start|요구 최소경력|0|모든 값이 0|
|6|<span style="color:orange">check_box_keyword|<span style="color:orange">모집직무코드|2507;2707;2810|; 으로 구분|
|7|<span style="color:orange">education|<span style="color:orange">요구학위|4||
|8|<span style="color:orange">major_task|<span style="color:orange">주업무코드|8||
|9|<span style="color:orange">qualifications|<span style="color:orange">자격요건난이도|1||
|10|text_keyword|모집직무키워드|MD;기획MD;여성복;캐주얼|; 으로 구분 / 707개의 데이터가 있지만 확인 결과 'check_box_keyword'컬럼이 더 객관적 분류 가능|
|11|<span style="color:orange">company_type_seq|<span style="color:orange">회사유형코드|2||
|12|<span style="color:orange">supply_kind|<span style="color:orange">주업종코드|514||
|13|<span style="color:orange">employee|<span style="color:orange">종업원수|20||
</details>

In [15]:
# 필요 데이터 추출
rec_data = pd.merge(recruitment_data.drop(['address_seq3','career_end','career_start','text_keyword'], axis='columns'), company_data, how='outer', on='recruitment_seq')
display(rec_data.tail(3))

recruitment_seq  address_seq1  address_seq2    check_box_keyword  \
6692          R03252           3.0           0.0                 2109   
6693          R05130           3.0           0.0  2201;2204;2205;2707   
6694          R04779           3.0           0.0  2101;2108;2201;2707   

      education  major_task  qualifications  company_type_seq  supply_kind  \
6692          3           2               1               4.0        402.0   
6693          2           2               2               2.0        402.0   
6694          3           8               1               2.0        402.0   

      employee  
6692     525.0  
6693      40.0  
6694      41.0

In [16]:
# 결측치가 있는 컬럼명 출력
print(', '.join([col for col in rec_data.columns if rec_data[col].isna().sum() != 0]))

company_type_seq, supply_kind, employee


In [17]:
# 데이터 정제 및 수정
rec_data.company_type_seq = rec_data.company_type_seq.replace(np.NAN, 0)
rec_data.supply_kind = rec_data.supply_kind.replace(np.NAN, 0)
rec_data.employee = rec_data.employee.replace(np.NAN, 0)

# 결측치 재확인
print(rec_data.isna().sum())

recruitment_seq      0
address_seq1         0
address_seq2         0
check_box_keyword    0
education            0
major_task           0
qualifications       0
company_type_seq     0
supply_kind          0
employee             0
dtype: int64


In [18]:
# 완성 데이터 저장
# rec_data.to_csv('./data/pre-processed_data/rec_data.csv', encoding='utf-8-sig')

### 1.3.2 이력서 관련 데이터 중 사용할 데이터 추출

<details>
<summary> <span style="color:gold"> 상세정보 펼쳐보기 </span> </summary>

</br>

|seq|columns|description|example|note|
|--|--|--|--|--|
|0|<span style="color:orange">resume_seq|<span style="color:orange">이력서번호|U01468|PK|
|1|reg_data|이력서등록일|2019-02-26||
|2|updated_data|이력서수정일|2020-06-23||
|3|<span style="color:orange">degree|<span style="color:orange">최종학력|4||
|4|graduate_data|졸업년도|2008||
|5|<span style="color:orange">hope_salary|<span style="color:orange">희망연봉|2100.0||
|6|last_salary|최종연봉|1000.0||
|7|<span style="color:orange">text_keyword|<span style="color:orange">직무키워드|상품기획MD;우븐디자인|; 으로 구분|
|8|<span style="color:green">job_code_seq1|<span style="color:green">희망직무1|경영·기획·회계·사무|희망직무 1~3은 순서에 상관없음으로 확인|
|9|<span style="color:green">job_code_seq2|<span style="color:green">희망직무2|재료·화학·섬유·의복|>> 희망직무1~3 를 기존의 12개의 직무를 바탕으로 새로운 분류 코드 생성|
|10|<span style="color:green">job_code_seq3|<span style="color:green">희망직무3|영업·판매·TM|>> '기타 직종'과 'NaN'은 '기타 직종' / '여행사'는 '관광'으로 통합|
|11|<span style="color:orange">career_month|<span style="color:orange">경력월|120|1년 단위로 그룹화|
|12|<span style="color:orange">career_job_code|<span style="color:orange">경력직무|섬유;봉제;가방;의류|; 으로 구분|
|13|<span style="color:orange">certificate_contents|<span style="color:orange">자격증명|주산2급|총 1787 종, 1인 기준 최대 10개 보유, 가나다순 자격증 리스트 작성|
|14|hischool_type_seq|고등학교유형코드|21|사용 X (최종학력으로만 구분 예정)|
|15|hischool_special_type|특수고등학교분류|일반고|사용 X (최종학력으로만 구분 예정)|
|16|hischool_nation|고등학교공립/사립구분|사립|사용 X (최종학력으로만 구분 예정)|
|17|hischool_gender|고등학교남녀/공학구분|여자학교|사용 X (최종학력으로만 구분 예정)|
|18|hischool_location_seq|고등학교지역코드|15|사용 X (최종학력으로만 구분 예정)|
|19|<span style="color:orange">univ_type_seq1|<span style="color:orange">최종대학(원)유형코드1|3|6종류|
|20|<span style="color:orange">univ_type_seq2|<span style="color:orange">최종대학(원)유형코드2|5|10종류|
|21|univ_tranfer|최종대학(원)편입여부|0|편입: 1|
|22|univ_location|최종대학(원)지역코드|15||
|23|univ_major|최종대학(원)전공|의류학과생활과학계열|주전공계열코드 사용|
|24|univ_sub_major|최종대학(원)부전공|의상디자인교육학|8482개의 데이터 중 416개의 데이터가 있지만, 전부 개별로 나뉘어져 사용 X|
|25|<span style="color:orange">univ_major_type|<span style="color:orange">최종대학(원)주전공계열코드|9||
|26|<span style="color:orange">univ_score|<span style="color:orange">최종대학(원)평균평점|80.0|0점, 1~10점, ... ,91~99점, 100점으로 그룹화|
|27|<span style="color:green">language|<span style="color:green">외국어코드|2|총 5종류|
|28|<span style="color:green">exam_name|<span style="color:green">공인시험코드|4|총 15종류|
|29|<span style="color:green">score|<span style="color:green">점수|883.84|가산점 형식으로 외국어 점수 계산; 점수에 따라 초급,중급,고급으로 분류(상대평가) 후 보유외국어 실력만큼 점수 추가|

</details>

#### 1.3.2.1 resume_data 중 사용할 데이터 추출

기존의 희망직무 코드 중 12개를 선정<br>

> 'IT·게임', '건설·기계·전기·전자','경영·기획·회계·사무', '관광', '기술·과학·산업', '디자인', '문화·예술·신문·방송', '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격', '통신·모바일', '기타 직종'

<br>
<span style="color:gold">희망직무 1 </span><br> 
**'IT·게임', '건설·기계·전기·전자', '경영·기획·회계·사무', '기타 직종', '디자인', '문화·예술·신문·방송', '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격'<br><br>
<span style="color:gold">희망직무 2 </span><br>  'IT·게임', <span style="color:pink">'NaN' >> '기타직종'</span>, '건설·기계·전기·전자', '경영·기획·회계·사무', '기술·과학·산업', '기타 직종', '디자인', '문화·예술·신문·방송', <span style="color:pink">'여행사' >> '관광'</span>, '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격'<br><br>
<span style="color:gold">희망직무 3 </span><br>  'IT·게임', <span style="color:pink">'NaN' >> '기타직종'</span>, '건설·기계·전기·전자', '관광', '기술·과학·산업', '기타 직종', '디자인', '문화·예술·신문·방송', '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격', '통신·모바일'

In [79]:
# 희망직무 통합 코드 만들기
## resume_data 중 사용할 컬럼 가져오기
resume_temp = resume_data[['resume_seq','degree','hope_salary','text_keyword','job_code_seq1','job_code_seq2','job_code_seq3','career_month','career_job_code']]

## 통합 12개의 희망직무 코드 중 선택한 직무 코드를 하나로 묶어주기
resume_temp = resume_temp.replace({'job_code_seq2':'여행사'},'관광').replace({'job_code_seq2':'NaN'},'기타 직종').replace({'job_code_seq3':'NaN'},'기타 직종')
new_job_code_seq = {'IT·게임':'a', '건설·기계·전기·전자':'b','경영·기획·회계·사무':'c', '관광':'d', '기술·과학·산업':'e', '디자인':'f', '문화·예술·신문·방송':'g', '영업·판매·TM':'h', '재료·화학·섬유·의복':'i', '전문·교육·자격':'j', '통신·모바일':'k', '기타 직종':''}

resume_temp['job_code_seq'] = ''
for i in range(len(resume_temp)):
    resume_temp.loc[i,'job_code_seq'] = ''.join(sorted([new_job_code_seq[resume_temp.loc[i]['job_code_seq1']], new_job_code_seq[resume_temp.loc[i]['job_code_seq2']],new_job_code_seq[resume_temp.loc[i]['job_code_seq3']]]))

## 기존의 희망직무1~3 제거
resume_temp = resume_temp.drop(['job_code_seq1','job_code_seq2','job_code_seq3'], axis='columns')



# 1000(만원)단위로 희망 연봉분류 
resume_temp['hope_salary_group'] = 0
for i in range(len(resume_temp)):
    resume_temp.loc[i,'hope_salary_group'] = int(resume_temp.loc[i]['hope_salary'] // 1000)

## 기존의 희망 연봉 컬럼 제거
resume_temp = resume_temp.drop(['hope_salary'], axis='columns')

# 데이터 확인
display(resume_temp.tail(5))

resume_seq  degree            text_keyword  career_month career_job_code  \
8477     U04890       5   상품기획;생산;글로벌소싱;남성복;여성복           204     섬유;봉제;가방;의류   
8478     U05910       4  소재;기획MD;남성복;온라인MD;디자이너            12     섬유;봉제;가방;의류   
8479     U01468       4            상품기획MD;우븐디자인             0     섬유;봉제;가방;의류   
8480     U05315       4    여성복디자인;우븐디자인;다이마루디자인            18      디자인;CAD;설계   
8481     U01277       4              상품기획;머천다이저             3     섬유;봉제;가방;의류   

     job_code_seq  hope_salary_group  
8477            i                  0  
8478            i                  2  
8479            f                  0  
8480            i                  0  
8481            i                  2

In [80]:
# 이력서 통합 데이터 (res_data)로 병합
res_data = resume_temp
res_data.tail(4)

resume_seq  degree            text_keyword  career_month career_job_code  \
8478     U05910       4  소재;기획MD;남성복;온라인MD;디자이너            12     섬유;봉제;가방;의류   
8479     U01468       4            상품기획MD;우븐디자인             0     섬유;봉제;가방;의류   
8480     U05315       4    여성복디자인;우븐디자인;다이마루디자인            18      디자인;CAD;설계   
8481     U01277       4              상품기획;머천다이저             3     섬유;봉제;가방;의류   

     job_code_seq  hope_salary_group  
8478            i                  2  
8479            f                  0  
8480            i                  0  
8481            i                  2

#### 1.3.2.2 resume_education_data 중 사용할 데이터 추출

In [81]:
# resume_education_data 중 필요한 컬럼 가져오기
resume_temp = resume_education_data[['resume_seq','univ_type_seq1','univ_type_seq2','univ_major_type','univ_score']]

# univ_score 그룹화
resume_temp['univ_score_group'] = 100
for i in range(len(resume_temp)):
    x = resume_temp.loc[i]['univ_score']
    if x == 0:
        resume_temp.loc[i,'univ_score_group'] = 0 
    else:
        resume_temp.loc[i,'univ_score_group'] = 1+(x-1)//10


## 기존의 점수 컬럼 제거
resume_temp = resume_temp.drop(['univ_score'], axis='columns')

# 데이터 확인
display(resume_temp.tail(5))

C:\Users\kimss\AppData\Local\Temp\ipykernel_13860\978520105.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  resume_temp['univ_score_group'] = 100


resume_seq  univ_type_seq1  univ_type_seq2  univ_major_type  \
8477     U06326               5               5               10   
8478     U07284               5               5                3   
8479     U04109               0               0                9   
8480     U01863               3               5                9   
8481     U05884               5               5                9   

      univ_score_group  
8477                 7  
8478                 7  
8479                 8  
8480                 8  
8481                 9

In [82]:
# 이력서 통합 데이터 (res_data)로 병합
res_data = pd.merge(res_data, resume_temp, how='outer', on='resume_seq')
res_data.tail(4)

resume_seq  degree            text_keyword  career_month career_job_code  \
8478     U05910       4  소재;기획MD;남성복;온라인MD;디자이너            12     섬유;봉제;가방;의류   
8479     U01468       4            상품기획MD;우븐디자인             0     섬유;봉제;가방;의류   
8480     U05315       4    여성복디자인;우븐디자인;다이마루디자인            18      디자인;CAD;설계   
8481     U01277       4              상품기획;머천다이저             3     섬유;봉제;가방;의류   

     job_code_seq  hope_salary_group  univ_type_seq1  univ_type_seq2  \
8478            i                  2               5               5   
8479            f                  0               5               5   
8480            i                  0               5               5   
8481            i                  2               5               5   

      univ_major_type  univ_score_group  
8478               19                 8  
8479                9                 8  
8480                9                 8  
8481                5                 8

#### 1.3.2.3 나머지 resume 관련 데이터 중 사용할 데이터 추출

> 많이 취득하는 자격증 통합

|번호|자격증|비고|
|--|--|--|
|1|자동차 운전면허 1종 보통||
|2|자동차 운전면허 2종 보통|자동차운전면허 포함|
|3|한자급수 자격시험|시험명 통일(급수는 그대로)|
|4|ITQ 정보기술자격|시험명 통일, 파워포인트 등 포함|
|5|MOS WORD|모스, 등 포함, (CORE, EXPERT 통합)|
|6|MOS EXCEL|마이크로소프트 엑셀 등 포함, (CORE, EXPERT 통합)|
|7|MOS POWERPOINT|MOS 파워포인트, MOS POWER POINT 등 포함|
|8|MOS ACCESS|CORE, EXPERT 통합|
|9|MOS OUTLOOK||
|10|MOS MASTER|MOSMASTER,모스마스터 등 포함, 세부 취득 기준은 제거|
|11|MCAS||
|12|워드프로세서|띄어쓰기 형식 통일|
|13|컴퓨터활용능력|컴활, 컴퓨터활용 등 포함, (1, 2급 통합)|

In [ ]:
resume_temp = resume_certificate_data
resume_temp

In [190]:
# resume_certificate_data.resume_seq.value_counts()
# len(resume_certificate_data.certificate_contents.value_counts())
resume_certificate_data.certificate_contents.value_counts().head(30)

자동차 운전면허 1종보통       563
운전면허2종보통            547
자동차운전면허 1종 보통       453
자동차 운전면허 (2종 보통)    441
자동차운전면허1종 보통        359
운전면허 1종 보통          299
1종보통 운전면허증          282
워드프로세서 2급           278
자동차운전면허 2종 보통       262
유통관리사 2급            254
컴퓨터그래픽스운용기능사        242
2종보통운전면허            230
워드프로세서 1급           207
컬러리스트 산업기사          167
샵마스터 3급             141
컬러리스트 기사            134
자동차 운전면허증 1종보통      130
GTQ 포토샵 1급          125
패션머천다이징산업기사         114
컴퓨터활용능력 2급          111
자동차운전면허             108
MOS MASTER          104
MOS MASTER 2003     102
자동차 운전 면허 1종         99
워드프로세서 3급            90
컬러리스트기사              87
1종보통운전면허증            84
MOS(엑셀               81
파워포인트                81
컬러리스트산업기사            81
Name: certificate_contents, dtype: int64

In [202]:
resume_certificate_data[resume_certificate_data.certificate_contents.str.contains('워드프로세서')]

resume_seq certificate_contents
34        U04397            워드프로세서 2급
50        U04876            워드프로세서 2급
68        U05093            워드프로세서 2급
94        U03938            워드프로세서 1급
143       U06247            워드프로세서 1급
...          ...                  ...
12890     U03056             워드프로세서2급
12897     U00686            워드프로세서 2급
12898     U01534            워드프로세서 1급
12920     U07040            워드프로세서 1급
12946     U08436            워드프로세서 2급

[601 rows x 2 columns]